db-tests/src/test/java/TestTriggers.java

```diff
@@ -759,4 +761,200 @@ SELECT COUNT(*) FROM dataset
+                );"""
+                .formatted(lastDatasetIdValue + 1)
+        )) {
+          res.next();
+          assertFalse(res.getBoolean("exists"));
+        }
+
+        insertDataset();
+
+        try (final var res = statement.executeQuery(
+            """
+                SELECT EXISTS(
+                  SELECT FROM information_schema.tables
+                  WHERE table_schema = 'public'
+                  AND table_name = 'profile_segment_%s'
+                );"""
+                .formatted(lastDatasetIdValue + 1)
+        )) {
+          res.next();
+          assertTrue(res.getBoolean("exists"));
+        }
+
+        try (final var res = statement.executeQuery(
+            """
+                SELECT EXISTS(
+                  SELECT FROM information_schema.tables
+                  WHERE table_schema = 'public'
+                  AND table_name = 'span_%s'
+                );"""
+                .formatted(lastDatasetIdValue + 1)
+        )) {
+          res.next();
+          assertTrue(res.getBoolean("exists"));
+        }
+
+        try (final var res = statement.executeQuery(
+            """
+                SELECT EXISTS(
+                  SELECT FROM information_schema.tables
+                  WHERE table_schema = 'public'
+                  AND table_name = 'event_%s'
+                );"""
+                .formatted(lastDatasetIdValue + 1)
+        )) {
+          res.next();
+          assertTrue(res.getBoolean("exists"));
+        }
+      }
+    }
+
+    @Test
+    void shouldDeletePartitionsOnDatasetDelete() throws SQLException {
+      try (final var statement = connection.createStatement()) {
+        try (final var res = statement.executeQuery(
+            """
+                SELECT EXISTS(
+                  SELECT FROM information_schema.tables
+                  WHERE table_schema = 'public'
+                  AND table_name = 'profile_segment_%s'
+                );"""
+                .formatted(datasetId)
+        )
+        ) {
+          res.next();
+          assertTrue(res.getBoolean("exists"));
+        }
+
+        try (final var res = statement.executeQuery(
+            """
+                SELECT EXISTS(
+                  SELECT FROM information_schema.tables
+                  WHERE table_schema = 'public'
+                  AND table_name = 'span_%s'
+                );"""
+                .formatted(datasetId)
+        )
+        ) {
+          res.next();
+          assertTrue(res.getBoolean("exists"));
+        }
+
+        try (final var res = statement.executeQuery(
+            """
+                SELECT EXISTS(
+                  SELECT FROM information_schema.tables
+                  WHERE table_schema = 'public'
+                  AND table_name = 'event_%s'
+                );"""
+                .formatted(datasetId)
+        )
+        ) {
+          res.next();
+          assertTrue(res.getBoolean("exists"));
+        }
+
+        insertDataset();
```

Comment:
Was this a copy/paste error? Why is a dataset being inserted here?

Reply:
Assert conditions prior to insert, do insert, assert conditions after insert